match_tpl.h

```diff
@@ -35,23 +33,57 @@ int32_t LONGEST_MATCH(deflate_state *const s, Pos cur_match) {
     ZLIB_REGISTER unsigned char *window = s->window;
     ZLIB_REGISTER unsigned char *scan = window + strstart;
     const Pos *prev = s->prev;
-    unsigned chain_length;
     Pos limit;
-    unsigned int best_len, nice_match;
-    bestcmp_t scan_end, scan_start;
+    uint32_t chain_length, nice_match, best_len, offset;
+    bestcmp_t scan_end;
+#ifndef UNALIGNED_OK
+    bestcmp_t scan_end0;
+#else
+    bestcmp_t scan_start;
+#endif
+
+#define EARLY_EXIT_TRIGGER_LEVEL 5
+
+#define RETURN_BEST_LEN \
+    if (best_len <= s->lookahead) \
```

Comment:
This if statement might not necessary since we already do such a check in `if (best_len >= nice_match) break`.